pkg/service/oidc4vp/oidc4vp_service.go

```diff
@@ -526,16 +540,17 @@ func (s *Service) createRequestObjectJWT(presentationDefinition *presexch.Presen
 		return "", fmt.Errorf("initiate oidc interaction: get key manager failed: %w", err)
 	}
 
-	vpFormats := getSupportedVPFormats(kms)
+	vpFormats := GetSupportedVPFormats(
+		kms.SupportedKeyTypes(), profile.Checks.Presentation.Format, profile.Checks.Credential.Format)
```

Comment:
vpFormats structure based on supported key types and verifier profile configuration